### ISSUE TITLE
Fix CssSyntaxError for interpolated variables in selector and flush opening brace

### ISSUE THREAD

**Issue by Tingglelaoo** [stylelint/stylelint#4829]
![image](https://user-images.githubusercontent.com/7029586/84588688-39502780-ae5c-11ea-8901-5b89624ef3a4.png)

It should be report the problem like this:
![image](https://user-images.githubusercontent.com/7029586/84588718-72889780-ae5c-11ea-9ef1-e801ff1f7598.png)

But when there is a less variable on selector, it cause the error report "Unknown word (CssSyntaxError)"

> Which rule, if any, is the bug related to?

> What code is needed to reproduce the bug?

```css
@prefix: t;

.@{prefix}-dialog{
  color: #fff;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "stylelint-scss",
    "stylelint-order",
  ],
  "rules": {
    "block-opening-brace-space-before": "always"
  }
}
```

> Which version of stylelint are you using?

13.5.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint **/*.{html,vue,css,wxss,less} --allow-empty-input`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to Less nested properties

> What did you expect to happen?

It should report the right problem.

> What actually happened (e.g. what warnings or errors did you get)?

"The following warnings were flagged:"

```shell
style/mobile/components/dialog/_index.less
 4:3  ✖  Unknown word   CssSyntaxError
```

**Comment by jeddy3** [comment]
@Tingglelaoo Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIACA3AhgJ3gAgGYQgG4AdAOwDolh0MBfYbSAGwiwxikO1pABoQAzgE9SAFzQAPBCCYwBAvuAilcASwDm04GWzZiIDAFdZA-Tm2ldO7mR60gA) with:

```less
@var: foo;
.@{var}{ color:red; }
```

It appears to be triggered when combining interpolation with a flush opening brace.

It's likely an issue with the parser stylelint uses [to supports Less](https://stylelint.io/about/syntaxes). Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes) if you have time to help fix the issue.

**Comment by likun7981** [comment]
This [demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIACBDAdgFwLQAcBOMAZgJYAe8ABOhgNwA6aDAdEsNbgSaQL5YYoAjADYw+ACxgooFFm0wciZXv2GiwMIUOAMKFSEIh5KBKPTTcGIADQgAzgE9MKUghAjbt6+AhoSAc1dtNF06EDwAV3dQyiDdEOCKC3MQbiA) will report the same error, I think I should is the cause of the general

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

Please consider fixing the issue upstream as the syntax is no longer bundled by Stylelint.
